### PR TITLE
[notifications] fix Expo Go compilation

### PR DESCRIPTION
**Pull request by vonovak**
# Why

this makes expo go build again. 

# How

There are a bunch of changed in gradle files. Most of them probably aren't needed but building locally and working on having a small diff is very time consuming due to build times; the biggest culprit was `com.google.android.gms.common.util.VisibleForTesting` which was included in `CampaignTrackingReceiver` which is the superclass of `InstallReferrerReceiver`. 

`CampaignTrackingReceiver` is from `com.google.android.gms:play-services-analytics` which is deprecated: https://developers.google.com/android/guides/setup. `InstallReferrerReceiver` injected `referrer` prop into the app - something that I did not find documented and the code was pretty old - so I removed the code for this (is there a changelog where I should put this?). Install referrer should be obtained via [Expo Application](https://docs.expo.dev/versions/latest/sdk/application/#applicationgetinstallreferrerasync) instead.

Also ran spotless which is where the unrelated changes come from.

# Test Plan

CI passes. I was also able to run the app locally and launch NCL.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: apps/expo-go/android/expoview/build.gradle [hunk 1]
```diff
@@ -183,45 +186,30 @@ dependencies {
...
   api 'com.squareup.picasso:picasso:2.5.2'
-  api 'com.google.android.gms:play-services-analytics:17.0.0'
+  // play-services-analytics is deprecated! (replaced with firebase)
+  api 'com.google.android.gms:play-services-analytics:18.0.1'
   api 'com.google.android.gms:play-services-maps:18.0.1'
-  api 'com.google.android.gms:play-services-auth:17.0.0'
+  api 'com.google.android.gms:play-services-auth:18.1.0'
   api 'com.google.android.gms:play-services-location:20.0.0'
   api 'com.google.android.gms:play-services-fitness:17.0.0'
-  api 'com.google.android.gms:play-services-wallet:17.0.0' //may need 10.+
```
Comment (gabrieldonadel) [inline apps/expo-go/android/expoview/build.gradle:191]: I believe we need for stripe

File: apps/expo-go/android/expoview/build.gradle [hunk 2]
```diff
@@ -183,45 +186,30 @@ dependencies {
...
   debugApi 'com.squareup.leakcanary:leakcanary-android-no-op:1.4-beta1'
   // debugApi 'com.squareup.leakcanary:leakcanary-android:1.4-beta1'
   releaseApi 'com.squareup.leakcanary:leakcanary-android-no-op:1.4-beta1'
   api 'commons-io:commons-io:2.6'
   api 'me.leolin:ShortcutBadger:1.1.4@aar'
-  implementation "com.github.CanHub:Android-Image-Cropper:4.3.0"
+  implementation "com.github.CanHub:Android-Image-Cropper:4.3.1"
   api 'commons-codec:commons-codec:1.10'
   api 'net.openid:appauth:0.7.1'
-  api 'com.airbnb.android:lottie:3.4.0'
```
Comment (gabrieldonadel) [inline apps/expo-go/android/expoview/build.gradle:200]: Don't we need this for lottie?

File: apps/expo-go/android/expoview/build.gradle (hunk 1 above)
Comment (vonovak) [reply apps/expo-go/android/expoview/build.gradle:191]: added back

File: apps/expo-go/android/expoview/build.gradle (hunk 2 above)
Comment (vonovak) [reply apps/expo-go/android/expoview/build.gradle:200]: added back

File: apps/expo-go/android/app/build.gradle [hunk 3]
```diff
@@ -219,24 +222,19 @@ dependencies {
...
-  implementation 'me.leolin:ShortcutBadger:1.1.4@aar'
-  implementation 'com.github.CanHub:Android-Image-Cropper:1.1.1'
+  implementation 'com.github.CanHub:Android-Image-Cropper:4.3.1'
   implementation 'commons-codec:commons-codec:1.10'
   implementation 'com.google.zxing:core:3.3.3'
   implementation 'net.openid:appauth:0.4.1'
-  implementation 'com.airbnb.android:lottie:3.4.0'
-  implementation('io.nlopez.smartlocation:library:3.2.11') {
-    transitive = false
-  }
```
Comment (gabrieldonadel) [inline apps/expo-go/android/app/build.gradle:239]: I believe this is used by expo-location, does it still works fine when removing this?

**Review by gabrieldonadel — COMMENTED** [review]
I think we might need to do a full check on NCL if we need to change all these dependencies to get expo go building

File: apps/expo-go/android/app/build.gradle (hunk 3 above)
Comment (vonovak) [reply apps/expo-go/android/app/build.gradle]: I added it back and build is passing; but I cant' tell for now if everything works with / without it bcs I cannot get the app to build.

Comment (vonovak) [reply apps/expo-go/android/app/build.gradle]: managed to build and test

File: apps/native-component-list/src/screens/ModulesCore/ModulesBenchmarksScreen.ios.tsx [hunk 4]
```diff
@@ -0,0 +1,216 @@
+import { useTheme } from 'ThemeProvider';
```
Comment (lukmccall) [inline apps/native-component-list/src/screens/ModulesCore/ModulesBenchmarksScreen.ios.tsx:1]: That screen should also work on Android. Why did you make it iOS only?

Comment (vonovak) [reply apps/native-component-list/src/screens/ModulesCore/ModulesBenchmarksScreen.ios.tsx:1]: interesting, for me, bundling was failing on android. Let me fix it

**Review by gabrieldonadel — COMMENTED** [review]
Can we break this PR up to only include the necessary changes? And then open other, smaller  PRs for the things you want to change? I understand that a lot of this is just code cleanup, but it would be better if we narrow down the exact problem that we're trying to fix here

File: apps/expo-go/android/expoview/src/main/java/host/exp/exponent/kernel/Kernel.kt [hunk 5]
```diff
@@ -343,18 +343,9 @@ class Kernel : KernelInterface() {
...
-      val exponentProps = JSONObject()
-      val referrer = exponentSharedPreferences.getString(ExponentSharedPreferences.ExponentSharedPreferencesKey.REFERRER_KEY)
-      if (referrer != null) {
-        try {
-          exponentProps.put("referrer", referrer)
-        } catch (e: JSONException) {
-          EXL.e(TAG, e)
-        }
-      }
       val bundle = Bundle()
```
Comment (Kudo) [inline apps/expo-go/android/expoview/src/main/java/host/exp/exponent/kernel/Kernel.kt:346]: the bundle is just empty bundle now, so we may able to refactor this code a little bit. e.g. `private val kernelLaunchOptions = Bundle()`

File: template-files/android/AndroidManifest.xml [hunk 6]
```diff
@@ -234,34 +234,6 @@
...
-              android:enabled="true">
-      <intent-filter>
-        <action android:name="com.google.android.gms.analytics.ANALYTICS_DISPATCH" />
-      </intent-filter>
-    </receiver>
-
-    <service android:name="com.google.android.gms.analytics.AnalyticsService"
-             android:enabled="true"
-             android:exported="false"/>
-
```
Comment (Kudo) [inline template-files/android/AndroidManifest.xml:264]: from your understanding, does the `AnalyticsService` and `AnalyticsReceiver` also relate to `InstallReferrerReceiver`?

File: apps/expo-go/android/expoview/build.gradle [hunk 7]
```diff
@@ -194,34 +190,20 @@ dependencies {
...
-    implementation('androidx.work:work-runtime:2.7.0') {
-      because '''androidx.work:work-runtime:2.1.0 pulled from
-      play-services-ads has a bug using PendingIntent without
-      FLAG_IMMUTABLE or FLAG_MUTABLE and will fail in Apps
-      targeting S+.'''
-    }
-  }
-
-  api 'com.google.firebase:firebase-core:21.1.0'
-  api 'com.google.firebase:firebase-messaging:22.0.0'
```
Comment (Kudo) [inline apps/expo-go/android/expoview/build.gradle:224]: we may also remove dep at https://github.com/expo/expo/blob/main/apps/expo-go/android/expoview/build.gradle#L236-L237 below

**Review by Kudo — COMMENTED** [review]
agree @gabrieldonadel to split the pr, something like
- dep updates
- spotlessApply
- VisibleTesting fix
- notification fix

from my study, the `InstallReferrerReceiver` and `referrer` props were originally for expo-go and expo-go's home analytics. it looks like we don't use the analytics anymore and it's safe to remove as you mentioned in the pr.

**Comment by vonovak** [comment]
superseded by https://github.com/expo/expo/pull/31736

**Pull request closed without merging**